lxd/storage/drivers/volume.go

```diff
@@ -145,23 +153,36 @@ func (v Volume) EnsureMountPath() error {
 		mode = os.FileMode(0100)
 	}
 
-	// Set mode of actual volume's mount path.
-	err = os.Chmod(volPath, mode)
+	fInfo, err := os.Lstat(volPath)
 	if err != nil {
-		return errors.Wrapf(err, "Failed to chmod '%s'", volPath)
+		return errors.Wrapf(err, "Error getting mount directory info %q", volPath)
+	}
+
+	// We expect the mount path to be a directory, so use this for comparison.
+	compareMode := os.ModeDir | mode
+
+	// Set mode of actual volume's mount path if needed.
+	if fInfo.Mode() != compareMode {
+		err = os.Chmod(volPath, mode)
+
+		// If the chmod failed, return the error as long as the volume is not a snapshot.
+		// If the volume is a snapshot, we must ignore the error as snapshots are readonly and cannot be
+		// modified after they are taken, such that any permission error is not fixable at mount time.
+		if err != nil && !v.IsSnapshot() {
```

Comment:
@stgraber in the end I decided to just skip chmod errors for snapshots, as any error that occurs here is not fixable, and we should just continue as best we can. For non-snapshot vols, any error here is critical and must return an error.

Reply:
Sounds good